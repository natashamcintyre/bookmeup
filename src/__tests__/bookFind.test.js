import React from 'react'
import BookFind from '../components/bookFind.js'

describe('bookSearch', () => {
  it('renders without crashing', () => {
    const component = mount(<BookFind />)
    expect(component).toMatchSnapshot()
  })

  it('has input textbox for ISBN', () => {
    const component = mount(<BookFind />)
    expect(component.exists('input#ISBNSearch')).toBe(true)
  })

  it('has submit button', () => {
    const component = mount(<BookFind />)
    expect(component.exists('button#search')).toBe(true)
  })

  it('should update state title when text entered', () => {
    const component = mount(<BookFind />)
    component.find('input#ISBNSearch').simulate('change', {
      target: { value: 'test_isbn' }
    })
    expect(component.state('isbn')).toEqual('test_isbn')
  })

  it('Clear message box on submit', () => {
    const component = mount(<BookFind submitISBN={ function (item) { return true } }/>)
    component.find('input#ISBNSearch').simulate('change', {
      target: { value: 'test_title' }
    })
    expect(component.state('isbn')).toEqual('test_title')
    component.find('form').simulate('submit')

    expect(component.find('input#ISBNSearch').props().value).toEqual('')
    expect(component.state('isbn')).toEqual('')
  })
})
