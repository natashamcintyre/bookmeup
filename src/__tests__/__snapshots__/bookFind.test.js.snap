// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bookSearch renders without crashing 1`] = `ReactWrapper {}`;
